juju/juju
# [JUJU-4012]  Watcher registry logger

Adds logging to the watcher registry so that it makes it easy to observe what the watchers are doing per connection.

Debugging a watcher from an apiserver is actually quite difficult if the underlying watcher dies and try to observe that from the logging perspective.

I added a new logging label to give a wider context to this.

## Checklist


## QA steps

```sh
$ juju bootstrap lxd test --build-agent
$ juju model-config -m controller logging-config="<root>=INFO;#watchers=TRACE"
```

Bounce the agents to see the pinger coming up.
